cmd/ctr/commands/run/run_unix.go

```diff
@@ -238,6 +238,7 @@ func NewContainer(ctx gocontext.Context, client *containerd.Client, context *cli
 				return nil, err
 			}
 			opts = append(opts, oci.WithAnnotations(annos))
+			opts = append(opts, oci.WithCDI(annos, []string{"/etc/cdi"}))
```

Comment:
If we're running from the command-line, why not give CDI devices their own command-line, e.g., `--cdi device,device,device`, like the existing [`--gpus`](https://github.com/containerd/containerd/blob/f7f2be732159a411eae46b78bfdb479b133a823b/cmd/ctr/commands/run/run_unix.go#L319-L321) or [`--device`](https://github.com/containerd/containerd/blob/f7f2be732159a411eae46b78bfdb479b133a823b/cmd/ctr/commands/run/run_unix.go#L333-L335) ([Windows](https://github.com/containerd/containerd/blob/f7f2be732159a411eae46b78bfdb479b133a823b/cmd/ctr/commands/run/run_windows.go#L154-L163)) flags?

It seems user-unkind to requiring the user to hand-build annotations just to get unparsed back into device names, particularly when the existing implementation is moving away from annotations towards using CDI device names in a dedicated field per #8252.

That would make `oci.WithCDI` take simply an array of device names (as strings) and allow it to be reusable underneath the CRI plugin's WithCDI too, unifying the place to deal with the CDI registry singleton.